packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts

```diff
@@ -1395,14 +1434,17 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 
 		// pause outgoing ops so we can detect dropped stashed changes
 		await container.deltaManager.outbound.pause();
-
 		let pendingState;
-		// always delete stash blob on "connected" event
-		container.on("connected", (clientId: string) => {
-			pendingState = container.getPendingLocalState?.();
+
+		const connectedEvent = async (clientId: string): Promise<void> => {
+			pendingState = await container.getPendingLocalState?.();
 			// the pending data in the stash blob may not have changed, but the clientId should match our new
 			// clientId, which will now be used to attempt to resubmit pending changes
 			assert.strictEqual(clientId, JSON.parse(pendingState).clientId);
+		};
+
+		container.on("connected", (clientId: string) => {
+			void connectedEvent(clientId);
```

Comment:
rather than voiding can you just hang on to the promise, and wait for it later? you can replace the setTimeout promise with awaiting this promise

Reply:
maybe make pendingState into pendingStateP and store the promise, which you await below

Reply:
I need pendingState in the event because of the assertion.

Reply:
I don't think the below comment is true anymore, which is bad:
```
// because the event listener was always refreshing pendingState on "connected", the stash blob
// should be safe to use
```
we don't clear it immediately on "connected" anymore, so it might be stale.